### PR TITLE
DATAREST-2087 Add support of path on RepositoryRestController

### LINKED ISSUE #2087: Feature regression : Spring Data REST controller must not use @RequestMapping on class level as this would cause double registration with Spring MVC
On latest version @RequestMapping isn't more allowed at class level due to this change =>

https://github.com/spring-projects/spring-data-rest/commit/46dc6e03fcd13cadfafeab22ba813c12a8dc9688

@RequestMapping is now only allowed on method level but that isn't an valid option for method on an abstract controller.
```
@RepositoryRestController
@RequestMapping("/api/widgets")
public class WidgetController  extends BaseControler {
    // ...
}

public class BaseControler {
    @GetMapping("/get")
    public ResponseEntity<String> get() {
        return ResponseEntity.ok().body(....);
    }
}
```

Proposition of fix =>

Add a field "path/value" on the annotation @RepositoryRestController to be able to set a root path at class level.

```
@RepositoryRestController("/api/widgets")
@RequiredArgsConstructor
public class WidgetController  extends BaseControler {
    // ...
}
public class BaseControler {
    @GetMapping("/get")
    public ResponseEntity<String> get() {
        return ResponseEntity.ok().body(....);
    }
}
```
Stackoverflow discussion =>
https://stackoverflow.com/questions/69825704/spring-data-rest-controller-must-not-use-requestmapping-on-class-level-as-this

### PR DESCRIPTION
**Pull request by yyvess**
Fix #2087 

Add a field "path/value" on the annotation @RepositoryRestController to be able to set a root path at class level.

### CODE REVIEW

**Comment by alienisty** [comment]
@RequestMappig supports an array of paths so I believe we should do the same for @RepositoryRestController

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/RepositoryRestController.java [hunk 1]
```diff
@@ -46,4 +47,9 @@
 @Target({ ElementType.TYPE, ElementType.ANNOTATION_TYPE })
 @BasePathAwareController
 public @interface RepositoryRestController {
+    @AliasFor("path")
+    String value() default "";
+
+    @AliasFor("value")
+    String path() default "";
```
Comment (alienisty) [inline spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/RepositoryRestController.java:54]: @RequestMapping allows specifying more than one path, should we do the same here?

Comment (yyvess) [reply spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/RepositoryRestController.java]: @alienisty I agree, I update it

**Comment by gauravbrills** [comment]
Hope this gets merged soon been a long time . For now we had to do a lot of refactor to avoid this in our latest migration :|

**Comment by abtabhi** [comment]
can we please triage this

**Comment by zampettim** [comment]
We are definitely running into this issue as well, and we have a ton of AbstractControllers with a lot of methods that we inherit. Gettting this fix into the 2.7.x line of Spring Boot would be extremely helpful.

**Comment by yyvess** [comment]
Commit of this PR was cherry-pick to main branch and included from the version S4.0.0-M5

**Comment by yyvess** [comment]
Merged by cherry-pick

**Pull request closed without merging**